snappi_tests/bgp/test_bgp_scalability.py fails with "Error in L2/L3 Traffic Apply"

### Title
snappi_tests/bgp/test_bgp_scalability.py fails with IxNetwork Error in L2/L3 Traffic Apply 

### Description
The test test_bgp_scalability.py fails during traffic start with IxNetwork apply error.

Observed behavior:
1. test_duthost_bgp_scalability_config passes.
2. test_bgp_scalability_16k_v4_routes fails.
3. Failure occurs when starting traffic control state and retrying in restart_traffic.
4. IxNetwork error reported repeatedly:
   Error in L2/L3 Traffic Apply

2 trunk Ethernet interfaces and 1 routed Portchannel interface on DUT:
```

snappi_tests/bgp/test_bgp_local_link_failover.py::test_bgp_convergence_for_local_link_failover[IPv4-1000-1-2] 
-------------------------------- live log setup --------------------------------
14/04/2026 07:08:51 snappi.api                               L0106 WARNING| Version check is disabled
ERROR                                                                    [100%]

==================================== ERRORS ====================================
_ ERROR at setup of test_bgp_convergence_for_local_link_failover[IPv4-1000-1-2] _
...
    @pytest.fixture(scope="module")
    def tgen_ports(duthost, conn_graph_facts, fanout_graph_facts):      # noqa: F811
        """
        Populate tgen ports info of T0 testbed and returns as a list
        Args:
            duthost (pytest fixture): duthost fixture
            conn_graph_facts (pytest fixture): connection graph
            fanout_graph_facts (pytest fixture): fanout graph
        Return:
            [{'card_id': '1',
            'ip': '22.1.1.2',
            'ipv6': '3001::2',
            'ipv6_prefix': u'64',
            'location': '10.36.78.238;1;2',
            'peer_device': 'sonic-s6100-dut',
            'peer_ip': u'22.1.1.1',
            'peer_ipv6': u'3001::1',
            'peer_port': 'Ethernet8',
            'port_id': '2',
            'prefix': u'24',
            'speed': 'speed_400_gbps'},
            {'card_id': '1',
            'ip': '21.1.1.2',
            'ipv6': '2001::2',
            'ipv6_prefix': u'64',
            'location': '10.36.78.238;1;1',
            'peer_device': 'sonic-s6100-dut',
            'peer_ip': u'21.1.1.1',
            'peer_ipv6': u'2001::1',
            'peer_port': 'Ethernet0',
            'port_id': '1',
            'prefix': u'24',
            'speed': 'speed_400_gbps'}]
        """
        speed_type = {
                      '10000': 'speed_10_gbps',
                      '25000': 'speed_25_gbps',
                      '40000': 'speed_40_gbps',
                      '50000': 'speed_50_gbps',
                      '100000': 'speed_100_gbps',
                      '200000': 'speed_200_gbps',
                      '400000': 'speed_400_gbps',
                      '800000': 'speed_800_gbps'}
        config_facts = duthost.config_facts(host=duthost.hostname,
                                            source="running")['ansible_facts']
        snappi_fanouts = get_peer_snappi_chassis(conn_data=conn_graph_facts,
                                                 dut_hostname=duthost.hostname)
        pytest_assert(snappi_fanouts is not None, 'Fail to get snappi_fanout')
        snappi_fanout_list = SnappiFanoutManager(fanout_graph_facts)
        for snappi_fanout in snappi_fanouts:
            snappi_fanout_id = list(fanout_graph_facts.keys()).index(snappi_fanout)
            snappi_fanout_list.get_fanout_device_details(device_number=snappi_fanout_id)
            snappi_ports = snappi_fanout_list.get_ports(peer_device=duthost.hostname)
            port_speeds = {int(p['speed']) for p in snappi_ports}
            if len(port_speeds) != 1:
                """ All the ports should have the same bandwidth """
                return None
            port_speed = port_speeds.pop()
            dutIps = create_ip_list(dut_ip_start, len(snappi_ports), mask=prefix_length)
            tgenIps = create_ip_list(snappi_ip_start, len(snappi_ports), mask=prefix_length)
            dutv6Ips = create_ip_list(dut_ipv6_start, len(snappi_ports), mask=v6_prefix_length)
            tgenv6Ips = create_ip_list(snappi_ipv6_start, len(snappi_ports), mask=v6_prefix_length)
            for port_id, port in enumerate(snappi_ports):
                port['speed'] = speed_type.get(str(port_speed), port['speed'])
                peer_port = port['peer_port']
>               int_addrs = list(config_facts['INTERFACE'][peer_port].keys())
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^
E               KeyError: 'INTERFACE'
```

3 routed (2 Ethernet and 1 Portchannel) interfaces on DUT:
```
snappi_tests/bgp/test_bgp_scalability.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
snappi_tests/bgp/files/bgp_test_gap_helper.py:123: in run_bgp_scalability_v4_v6
    get_bgp_scalability_result(snappi_api, localhost, tgen_bgp_config, limit_flag, duthost)
        dual_stack_flag = 1
        duthost    = <MultiAsicSonicHost DUT hostname redacted>
        ipv4_routes = 16000
        ipv6_prefix = 64
        ipv6_routes = 1
        limit_flag = 0
        localhost  = <tests.common.devices.local.Localhost object at 0x7f2a778d93a0>
        multipath  = 1
        port_count = 2
        snappi_api = <snappi_ixnetwork.snappi_api.Api object at 0x7f2a6c445070>
        tgen_bgp_config = <snappi.snappi.Config object at 0x7f2a783904a0>
        tgen_ports = [{'card_id': '2', 'ip': '20.1.1.1', 'ipv6': '2000:1::2', 'ipv6_prefix': 126, ...}, {'card_id': '2', 'ip': '20.1.1.3', ...000:1::6', 'ipv6_prefix': 126, ...}, {'card_id': '2', 'ip': '1.0.0.57', 'ipv6': 'fc00::71', 'ipv6_prefix': '126', ...}]
snappi_tests/bgp/files/bgp_test_gap_helper.py:704: in get_bgp_scalability_result
    warning = run_traffic(snappi_api, duthost)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bgp_config = <snappi.snappi.Config object at 0x7f2a783904a0>
        duthost    = <MultiAsicSonicHost DUT hostname redacted>
        flag       = 0
        ixnet      = <ixnetwork_restpy.testplatform.sessions.ixnetwork.ixnetwork.Ixnetwork object at 0x7f2a66def2c0>
        localhost  = <tests.common.devices.local.Localhost object at 0x7f2a778d93a0>
        snappi_api = <snappi_ixnetwork.snappi_api.Api object at 0x7f2a6c445070>
snappi_tests/bgp/files/bgp_test_gap_helper.py:665: in run_traffic
    restart_traffic(snappi_api)
        cs         = <snappi.snappi.ControlState object at 0x7f2a79287830>
        duthost    = <MultiAsicSonicHost DUT hostname redacted>
        snappi_api = <snappi_ixnetwork.snappi_api.Api object at 0x7f2a6c445070>
        var        = "2026 Apr 11 07:32:36.362732 DUT hostname redacted INFO python3.13[1272328]: ansible-ansible.legacy.command Invoked with...stdin_add_newline=True strip_empty_ends=True argv=None chdir=None executable=None creates=None removes=None stdin=None"
        warning    = 1
snappi_tests/bgp/files/bgp_test_gap_helper.py:644: in restart_traffic
    snappi_api.set_control_state(cs)
        cs         = <snappi.snappi.ControlState object at 0x7f2a65fb7f10>
        snappi_api = <snappi_ixnetwork.snappi_api.Api object at 0x7f2a6c445070>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <snappi_ixnetwork.snappi_api.Api object at 0x7f2a6c445070>
payload = <snappi.snappi.ControlState object at 0x7f2a65fb7f10>

    def set_control_state(self, payload):
        try:
            control_option = payload.choice
            control_obj = getattr(payload, control_option)
            control_choice = control_obj.get("choice")
            request_payload = getattr(control_obj, control_choice)
            self._connect()
            event_names = []
            event_state = None
            event_type = control_choice
            EventInfo = namedtuple(
                "EventInfo", ["event_type", "event_state", "event_names"]
            )
            if control_option == "port":
                if control_choice == "capture":
                    self.capture.set_capture_state(request_payload)
                elif control_choice == "link":
                    self._TRIGGERED_EVENT = "link"
                    if request_payload.port_names is not None:
                        event_names = request_payload.port_names
                        event_state = request_payload.state
                    self.vport.set_link_state(request_payload)
            elif control_option == "protocol":
                if control_choice == "all":
                    self.ngpf.set_protocol_state(request_payload)
                elif control_choice == "route":
                    event_state = request_payload.state
                    with Timer(self, "Setting route state"):
                        event_names = self.ngpf.set_route_state(
                            request_payload
                        )
                elif control_choice == "lacp":
                    self.ngpf.set_device_state(request_payload)
            elif control_option == "traffic":
                self.traffic_item.transmit(request_payload)
            elif control_option is not None:
                msg = "{} is not a supported choice for metrics; \
                the supported choices are \
                ['port', 'protocol', traffic]".format(
                    control_option
                )
                raise SnappiIxnException(400, msg)
            self._event_info = EventInfo(event_type, event_state, event_names)
        except Exception as err:
>           raise SnappiIxnException(err)
E           snappi_ixnetwork.exceptions.SnappiIxnException:   File "/opt/venv/lib/python3.12/site-packages/snappi_ixnetwork/snappi_api.py", line 477, in set_control_state
E               self.traffic_item.transmit(request_payload)
E             File "/opt/venv/lib/python3.12/site-packages/snappi_ixnetwork/trafficitem.py", line 1619, in transmit
E               self._api._traffic.Apply()
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/testplatform/sessions/ixnetwork/traffic/traffic.py", line 1428, in Apply
E               return self._execute("apply", payload=payload, response_object=None)
E                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/base.py", line 452, in _execute
E               response = self._connection._execute(url, payload)
E                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/connection.py", line 308, in _execute
E               return self._send_recv("POST", url, payload)
E                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/connection.py", line 591, in _send_recv
E               return self._poll()
E                      ^^^^^^^^^^^^
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/connection.py", line 659, in _poll
E               return self._process_response_status_code(
E                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E             File "/opt/venv/lib/python3.12/site-packages/ixnetwork_restpy/connection.py", line 495, in _process_response_status_code
E               raise BadRequestError(message, response.status_code)
E             Error in L2/L3 Traffic Apply
```



### Environment
- Repository: sonic-net/sonic-mgmt
- Python: 3.12.3
- pytest: 9.0.2
- ansible: 2.18.14

### Repro Command
python3 -m pytest snappi_tests/bgp/test_bgp_scalability.py --inventory <inv> --host-pattern <DUT> --dpu-pattern None --testbed <testbed name> --testbed_file <testbed yaml> --log-cli-level warning --log-file-level debug --kube_master unset --showlocals --assert plain --show-capture no -rav --allow_recover --ignore=ptftests --ignore=acstests --ignore=saitests --ignore=scripts --ignore=k8s --ignore=sai_qualify --log-file logs/snappi_tests/bgp/test_bgp_scalability.log --junitxml=logs/snappi_tests/bgp/test_bgp_scalability.xml -s --trim_inv --skip_sanity --disable_loganalyzer --pdb

### Key Trace Locations
- test_bgp_scalability.py
- bgp_test_gap_helper.py

### Error Summary
- ixnetwork_restpy.errors.BadRequestError: Error in L2/L3 Traffic Apply
- snappi_ixnetwork.exceptions.SnappiIxnException raised from set_control_state during flow_transmit start

### Notes
Potential mismatch between DUT-side interface model and TGEN-side L2/L3 object model may be causing IxNetwork traffic apply to fail.

### Impact
BGP scalability suite cannot proceed beyond the first traffic test case on this testbed.

`

--- a/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
+++ b/tests/common/plugins/conditional_mark/tests_mark_conditions.yaml
@@ -4842,6 +4842,42 @@ snappi_tests:
     conditions:
       - "asic_type in ['vs']"
 
+snappi_tests/bgp/test_bgp_convergence_performance.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
+snappi_tests/bgp/test_bgp_local_link_failover.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
+snappi_tests/bgp/test_bgp_remote_link_failover.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
+snappi_tests/bgp/test_bgp_rib_in_capacity.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
+snappi_tests/bgp/test_bgp_rib_in_convergence.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
+snappi_tests/bgp/test_bgp_scalability.py:
+  xfail:
+    reason: "IxNetwork Error in L2/L3 Traffic Apply - https://github.com/sonic-net/sonic-mgmt/issues/23857"
+    conditions:
+      - "https://github.com/sonic-net/sonic-mgmt/issues/23857"
+
 snappi_tests/dash:
   skip:
     reason: "Skipping dash on 2025 releases"

